src/main/java/taxi/dao/jdbc/CarDaoJdbc.java

```diff
@@ -0,0 +1,201 @@
+        driver.setId(driverId);
+        return driver;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars SET manufacturer_id = ?, model = ? "
+                + "WHERE car_id = ? AND deleted = false";
+        String deleteDriversQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String insertDriversQuery = "INSERT INTO cars_drivers(driver_id, car_id) VALUES (?, ?)";
+        Long carId = car.getId();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStmt = connection.prepareStatement(updateCarQuery);
+                PreparedStatement deleteDriversStmt =
+                        connection.prepareStatement(deleteDriversQuery);
+                PreparedStatement insertDriversStmt =
+                        connection.prepareStatement(insertDriversQuery)) {
+
+            updateCarStmt.setLong(1, car.getManufacturer().getId());
+            updateCarStmt.setString(2, car.getModel());
+            updateCarStmt.setLong(3, carId);
+            updateCarStmt.executeUpdate();
+
+            deleteDriversStmt.setLong(1, carId);
+            deleteDriversStmt.executeUpdate();
+
+            for (Driver driver : car.getDrivers()) {
+                insertDriversStmt.setLong(1, driver.getId());
+                insertDriversStmt.setLong(2, carId);
+                insertDriversStmt.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not update " + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = true WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement = connection.prepareStatement(query)) {
+
+            deleteStatement.setLong(1, id);
+            int deletedRows = deleteStatement.executeUpdate();
+            return deletedRows != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getFiltered = "SELECT cars.car_id AS car_id, cars.model, m.manufacturer_id AS m_id, "
+                + "m.manufacturer_name AS m_name, m.manufacturer_country AS m_country, "
+                + "d.driver_id AS d_id, d.driver_name AS d_name, d.license_number FROM cars "
+                + "LEFT JOIN cars_drivers cd ON cars.car_id = cd.car_id "
+                + "LEFT JOIN drivers d ON d.driver_id = cd.driver_id "
+                + "LEFT JOIN manufacturers m ON cars.manufacturer_id = m.manufacturer_id "
+                + "WHERE cars.deleted = FALSE AND d.deleted = false "
+                + "AND cars.car_id IN (SELECT car_id FROM cars_drivers WHERE driver_id = ?)";
```

Comment:
that makes this query really complicated. But let's leave it as it is since you understand the pros and cons